bigquery_storage/tests/unit/test_reader.py

```diff
@@ -337,6 +412,74 @@ def test_to_dataframe_w_dtypes(class_under_test):
     )
 
 
+def test_to_dataframe_by_page(class_under_test, mock_client):
+    bq_columns = [
+        {"name": "int_col", "type": "int64"},
+        {"name": "bool_col", "type": "bool"},
+    ]
+    avro_schema = _bq_to_avro_schema(bq_columns)
+    read_session = _generate_read_session(avro_schema)
+    bq_blocks_1 = [
```

Comment:
Minor nit:  the naming of the "blocks" variables reads oddly to me.  You're testing a single block with multiple rows, not multiple blocks.   It's consistent with existing tests, just seems odd.

Reply:
Actually, this does contain 2 blocks. `black` just formatted it in such a way that it's not clear.

I've separated each block into its own variable and added a comment for why there are two groups of blocks. Hopefully this makes it clearer.